04-onion-routing.md

```diff
@@ -629,6 +629,114 @@ Otherwise, the node sending the error is an intermediate node:
 The origin node MAY use the data specified in the various types of
 failure for debugging purposes.
 
+## Test Vector
+
+### Packet Creation
+
+The following is an in-depth trace of the packet creation, including intermediate data.
+
+
+#### Parameters
+
+	pubkey[0] 0x02eec7245d6b7d2ccb30380bfbe2a3648cd7a942653f5aa340edcea1f283686619
+	pubkey[1] 0x0324653eac434488002cc06bbfb7f10fe18991e35f9fe4302dbea6d2353dc0ab1c
+	pubkey[2] 0x027f31ebc5462c1fdce1b737ecff52d37d75dea43ce11c74d25aa297165faa2007
+	pubkey[3] 0x032c0b7cf95324a07d05398b240174dc0c2be444d96b159aa6c7f7b1e668680991
+	pubkey[4] 0x02edabbd16b41c8371b92ef2f04c1185b4f03b6dcd52ba9b78d9d7c89c8f221145
+
+	nhops = 5/20
+	sessionkey = 0x4141414141414141414141414141414141414141414141414141414141414141
+	associated data = 0x4242424242424242424242424242424242424242424242424242424242424242
+
+	hop_payload[0] = 0x4141414141414141414141414141414141414141
+	hop_payload[1] = 0x4141414141414141414141414141414141414141
+	hop_payload[2] = 0x4141414141414141414141414141414141414141
+	hop_payload[3] = 0x4141414141414141414141414141414141414141
+	hop_payload[4] = 0x4141414141414141414141414141414141414141
+
+#### Per-Hop Information
+
+	hop_shared_secret[0] = 0xa6254636b0f9fe6a871ef7057de6b1fa0f31e6f0e46990747fda49f7871a9f1f
```

Comment:
Otherwise I get the same results. Thanks!
By the way, if think that there is a typo in the specs: line 254 reads: "... The blinding factor is the SHA256 hash of the concatenation between the node's public key nodepk_k and ...".  Shouldn't it be the ephemereal public key instead of the node's public key ?